internal/core/project.go

```diff
@@ -245,6 +245,61 @@ func (p *Project) Ref() *pb.Ref_Project {
 	return &pb.Ref_Project{Project: p.name}
 }
 
+// Copy creates a copy of the project, for a different workspace.
+// The project's config is required to be passed in because the Config
+// option is not set on a project, so we can't reference it directly.
+// Getters for other project fields are used here to limit their exposure.
+func (p *Project) Copy(ctx context.Context, workspace string, projConfig *config.Config) (*Project, error) {
```

Comment:
I'd probably call this `InWorkspace` rather than `Copy` to convey better why it would be used.